src/Controller/Component/SecurityComponent.php

```diff
@@ -321,7 +318,7 @@ protected function _authRequired(Controller $controller)
      */
     protected function _validatePost(Controller $controller)
     {
-        if (empty($controller->request->data)) {
+        if (empty($controller->request->data())) {
```

Comment:
Here, we could use the new ``hasData()`` method. Refs: https://github.com/cakephp/cakephp/pull/9551/files#diff-c1445fb25adc45b0f569da99d2b2e7e8R112